### PR TITLE
Plugins/braintree

### PR DESCRIPTION
**Pull request by TheGrinch00**
Ho creato delle funzioni per:
- Customer
  - Create
  - Find
  - Delete
- PaymentMethod
  - Create
  - getFirst
- Subscription
  - Create
  - isAlreadySubscribed
  - Get all
- Discounts
  - Get all
  - Find by name

### CODE REVIEW

File: packages/plugins/braintree/src/braintree/index.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    number: string,
+    expirationMonth: string,
+    expirationYear: string,
+}
+
+class Braintree {
+    gateway: BraintreeGateway;
+
+    init() {
+        const environment = process.env.RLN_IS_BRAINTREE_SANDBOX ? braintreeSdk.Environment.Sandbox : braintreeSdk.Environment.Production;
```
Comment (antoniogiordano) [inline packages/plugins/braintree/src/braintree/index.ts:28]:
```suggestion
        const environment = process.env.RLN_BRAINTREE_IS_SANDBOX ? braintreeSdk.Environment.Sandbox : braintreeSdk.Environment.Production;
```

**Review by antoniogiordano — CHANGES_REQUESTED** [review]

File: packages/plugins/braintree/src/braintree/index.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    BraintreeGateway,
+    Customer,
+    CustomerCreateRequest, Discount,
+    GatewayConfig,
+    PaymentMethod, Plan,
+    Subscription,
+    ValidatedResponse,
+} from 'braintree';
+
+require('dotenv').config();
```
[the remark this answers is not in the document]
Comment (TheGrinch00) [reply packages/plugins/braintree/src/braintree/index.ts:12]: Without this, my keys were undefined

[the remark this answers is not in the document]
Comment (antoniogiordano) [reply packages/plugins/braintree/src/braintree/index.ts:12]: got it, but from the project env vars will be loaded with automatically, so we can remove from here now

Comment (antoniogiordano) [inline packages/plugins/braintree/src/braintree/index.ts:12]:
```suggestion
```

**Review by antoniogiordano — APPROVED** [review]
Awesome, let's try it! As soon as we connect it to a project, we'll realize if we need to add or modify anything!

**Pull request merged**